pontoon/base/placeables.py

```diff
@@ -0,0 +1,45 @@
+import re
+
+
+class PythonFormatNamedString:
+    parser = r"(%\([[\w\d!.,[\]%:$<>+\-= ]*\)[+|-|0\d+|#]?[.\d+]?[s|d|e|f|g|o|x|c|%])"
+
+
+class PythonFormatString:
+    parser = r"(\{{?[\w\d!.,[\]%:$<>+-= ]*\}?})"
+
+
+class PythonFormattingVariable:
+    parser = r"(%(%|(\([^)]+\)){0,1}[-+0#]{0,1}(\d+|\*){0,1}(\.(\d+|\*)){0,1}[hlL]{0,1}[diouxXeEfFgGcrs]{1}))"
+    match_index = 0
+
+
+class FluentTerm:
+    parser = r"({ ?-[^}]* ?})"
+
+
+class FluentParametrizedTerm:
+    parser = r"({ ?-[^}]*([^}]*: ?[^}]*) ?})"
+
+
+class FluentFunction:
+    parser = r"({ ?[A-W0-9\-_]+[^}]* ?})"
+
+
+def get_placeables(text):
+    """Return a list of placeables found in the given string."""
+    parsers = [
+        PythonFormatNamedString.parser,
+        PythonFormatString.parser,
+        # PythonFormattingVariable.parser,
+        FluentTerm.parser,
+        # FluentParametrizedTerm.parser,
+        FluentFunction.parser,
+    ]
+
+    placeables = []
+
+    for parser in parsers:
+        placeables += re.findall(parser, text)
```

Comment:
As `re.findall` doesn't care about groups, the `(...)` around the expressions above could be dropped.

Reply:
To correct myself, [re.findall](https://docs.python.org/3/library/re.html#re.findall) _does_ care about groups, but it treats having one capturing group for the whole expression exactly the same as not having any capturing groups.